gitlabform/gitlabform/processors/util/branch_protector.py

```diff
@@ -32,6 +32,11 @@ def protect_branch(self, project_and_group, configuration, branch):
                     and "merge_access_level"
                     and "unprotect_access_level"
                 ) in configuration["branches"][branch]:
+                    branch_access_levels = self.gitlab.get_branch_access_levels(project_and_group, branch)
+                    levels = ["push_access_levels", "merge_access_levels", "unprotect_access_levels"]
+                    if all(configuration["branches"][branch][level[0:-1]] == branch_access_levels[level][0]["access_level"] for level in levels):
```

Comment:
Erm... What? 😅 
Can you either simplify this code (preferably) or add a comment explaining what is happening here?

Reply:
A friendly reminder, @YuraBeznos. :)

Reply:
Hey, hello. Sorry for a late response. Looks like I will find time to fix tests somewhere in summer. I do remember about this but slightly busy with other things.

Reply:
Is the following comment enough to explain what the `if all(..)` is doing?
```
# Check each access type has the correct access level, if they do, not change is needed
# Gitlabform uses access_levels with plural form, and gitlab uses singular form.
# [0:-1] removes the plural
```

Reply:
Yes, that comment is enough, thanks @robin-moss! ...but it's the other way around, it's GitLab that uses plural (see https://docs.gitlab.com/ee/api/protected_branches.html#list-protected-branches) while GitLabForm uses singular.

Reply:
I've updated this also. Thanks.